nrpy/infrastructures/BHaH/general_relativity/BSSN/psi4_decomposition.py

```diff
@@ -3,161 +3,542 @@
 
 Author: Zachariah B. Etienne
         zachetie **at** gmail **dot* com
+        Nishita Jadoo
+        njadoo **at** uidaho **dot* edu
 """
 
 import nrpy.c_function as cfc
+import nrpy.params as par
+from nrpy.infrastructures.BHaH import BHaH_defines_h, griddata_commondata
+from nrpy.infrastructures.superB.CurviBoundaryConditions import (
+    register_CFunction_apply_bcs_inner_only_specific_gfs,
+    register_CFunction_apply_bcs_outerextrap_and_inner_specific_gfs,
+)
 
+par.register_CodeParameter(
+    "int", __name__, "num_theta_points_on_shell_for_psi4_interp", 32, commondata=True
+)
 
-def register_CFunction_psi4_spinweightm2_decomposition_on_sphlike_grids() -> None:
-    """Register C function for decomposing psi4 into spin-weighted spherical harmonics."""
-    prefunc = r"""
-static void lowlevel_decompose_psi4_into_swm2_modes(const int Nxx_plus_2NGHOSTS1,const int Nxx_plus_2NGHOSTS2,
-                                                    const REAL dxx1, const REAL dxx2,
+# Define governing parameters for extraction radii (R_exts)
+R_ext_min, R_ext_max, num_R_exts = 36.923, 150.0, 8
```

Comment:
Globals like this are not allowed in NRPy.

Reply:
Please remove lines 23-33 in favor of basic default values when registering the CodeParameters below. These can and should be adjusted via e.g., 

```
    par.adjust_CodeParam_default("num_psi4_extraction_radii", 1)
    par.adjust_CodeParam_default("list_of_psi4_extraction_radii", [80.0])
```
in the relevant e.g., NRPy example script.